maven-model-builder/src/main/java/org/apache/maven/model/building/DefaultModelBuilder.java

```diff
@@ -444,6 +461,64 @@ else if ( !parentIds.add( parentData.getId() ) )
         return result;
     }
 
+    private Map<String, Activation> getInterpolatedActivations( Model rawModel,
+                                                                DefaultProfileActivationContext context,
+                                                                DefaultModelProblemCollector problems )
+    {
+        Map<String, Activation> interpolatedActivations = getProfileActivations( rawModel, true );
+        for ( Activation activation : interpolatedActivations.values() )
+        {
+            if ( activation.getFile() != null )
+            {
+                replaceWithInterpolatedValue( activation.getFile(), context, problems );
+            }
+        }
+        return interpolatedActivations;
+    }
+
+    private void replaceWithInterpolatedValue( ActivationFile file, ProfileActivationContext context,
+                                               DefaultModelProblemCollector problems  )
+    {
+        String path;
+        boolean missing;
+
+        if ( StringUtils.isNotEmpty( file.getExists() ) )
```

Comment:
This is weird. Why does getExists return a string?

Also, we shouldn't need StringUtils any more in java 7.

Reply:
I rename `file` -> `activationFile` to make it clear.  Therefore `activationFile.getExists()` or `activationFile.getMissing()` return `String`.

`StringUtils.isNotEmpty` checks if the string is `null` or empty. 

@elharo Does it make sense for you now?

Reply:
We shouldn't be using this class at all. Prefer org.apache.commons.lang3.StringUtils though even that is overkill here IMHO

Reply:
@elharo I can replace it with 
```
if ( activationFile.getExists()  != null && !activationFile.getExists().isEmpty() )
```
If you wish, but for me `isNotEmpty()` is more cleaner